app/src/main/java/com/greenfox/kalendaryo/SelectCalendarActivity.java

```diff
@@ -87,23 +96,50 @@ public void getCalendarList() {
         ArrayList<String> accounts = kalPref.getAccounts();
 
         for (int i = 0; i < accounts.size(); i++) {
+            CURRENT_ATTEMPT = FIRST_ATTEMPT;
             GoogleAuth googleAuth = kalPref.getAuth(accounts.get(i));
-
-            String accessToken = googleAuth.getAccessToken();
-            String authorization = "Bearer " + accessToken;
-
-            googleApi.getCalendarList(authorization).enqueue(new Callback<GoogleCalendarsResponse>() {
-                @Override
-                public void onResponse(Call<GoogleCalendarsResponse> call, Response<GoogleCalendarsResponse> response) {
+            requestCalendars(googleAuth);
+        }
+    }
+    public void requestCalendars (GoogleAuth googleAuth) {
+        String accessToken = googleAuth.getAccessToken();
+        String authorization = "Bearer " + accessToken;
+        googleApi.getCalendarList(authorization).enqueue(new Callback<GoogleCalendarsResponse>() {
+            @Override
+            public void onResponse(Call<GoogleCalendarsResponse> call, Response<GoogleCalendarsResponse> response) {
+                if (response.errorBody() == null) {
                     adapter.addGoogleCalendars(response.body().getItems());
                     googleCalendars.addAll(response.body().getItems());
+                } else if (CURRENT_ATTEMPT != FINAL_ATTEMPT){
+                    requestAccessTokenRefresh(googleAuth);
                 }
+            }
+
+            @Override
+            public void onFailure(Call<GoogleCalendarsResponse> call, Throwable t) {
+                t.printStackTrace();
+            }
+        });
+    }
 
-                @Override
-                public void onFailure(Call<GoogleCalendarsResponse> call, Throwable t) {
-                    t.printStackTrace();
+    public void requestAccessTokenRefresh (GoogleAuth googleAuth) {
```

Comment:
refresh token mechanism should be general and not tied to the `requestCalendar` feature
you should move it to an other class and it should not invoke a specific method (in this example `requestCalendars(googleAuth);`